python/api/job/views.py

```diff
@@ -124,31 +124,40 @@ def get_task_download_archive_file_handler(task: TaskLog) -> io.BytesIO:
     jobs = JobLog.objects.filter(task=task)
 
     if task.action and task.action.type == ActionType.Job:
-        dir_name_suffix = str_remove_non_alnum(
+        task_dir_name_suffix = str_remove_non_alnum(
             value=task.action.display_name
         ) or str_remove_non_alnum(value=task.action.name)
     else:
-        dir_name_suffix = None
+        task_dir_name_suffix = None
 
     fh = io.BytesIO()
     with tarfile.open(fileobj=fh, mode="w:gz") as tar_file:
         for job in jobs:
-            if dir_name_suffix is None:
-                dir_name_suffix = str_remove_non_alnum(
-                    value=job.sub_action.action.display_name
-                ) or str_remove_non_alnum(value=job.sub_action.action.name)
+            if task_dir_name_suffix is None:
+                dir_name_suffix = ""
+                if job.sub_action:
+                    dir_name_suffix = str_remove_non_alnum(
+                        value=job.sub_action.display_name
+                    ) or str_remove_non_alnum(value=job.sub_action.name)
+            else:
+                dir_name_suffix = task_dir_name_suffix
 
             directory = Path(settings.RUN_DIR, str(job.pk))
             if directory.is_dir():
-                for log_file in Path(settings.RUN_DIR, str(job.pk)).iterdir():
-                    tarinfo = tarfile.TarInfo(f"{job.pk}-{dir_name_suffix}/{log_file.name}")
+                for log_file in [
+                    item for item in Path(settings.RUN_DIR, str(job.pk)).iterdir() if item.is_file()
+                ]:
+                    tarinfo = tarfile.TarInfo(
+                        f"{job.pk}-{dir_name_suffix}".strip("-") + f"/{log_file.name}"
```

Comment:
Use pls f-string, do not use `+`

Reply:
I need to use strip() to delete extra "-" if dir_name_suffix is absent (It is possible if subaction is None and bundle deleted). I can't strip the whole string and alternate variant is ternar operator and checking dir_name_suffix. I think way with "+" better and shorter.

Reply:
You can use:
```suggestion
                        f'{f"{job.pk}-{dir_name_suffix}".strip("-")}/{log_file.name}'
```
or:
```suggestion
                        str(Path(f"{job.pk}-{dir_name_suffix}".strip("-"), f"{log_file.name}"))
```

Reply:
Done